frontends/web/src/routes/account/add/addaccount.tsx

```diff
@@ -50,7 +51,18 @@ class AddAccount extends Component<TranslateProps, State> {
             accountName: this.state.accountName,
             extendedPublicKey: this.state.extendedPublicKey,
         };
-        apiPost('account-add', body).then(accountCode => route('/account/' + accountCode));
+        interface ResponseData {
+            success: boolean;
+            errorCode?: 'xpubInvalid' | 'xpubWrongNet';
+            accountCode?: string;
+        }
```

Comment:
If you want to improve/exercise your TypeScript skills, you could write a [user-defined type guard](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards) to distinguish between the successful and the unsuccessful format. 😉 But I agree it's almost a bit overkill in this case.

Reply:
TIL. Thanks.